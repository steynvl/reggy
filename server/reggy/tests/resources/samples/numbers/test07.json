{
  "sampleStringsInfo": [
    [
      {
        "markerType": "Numbers",
        "markedStrings": [
          "adsads"
        ],
        "markerInfo": {
          "minValOfIntPart": 0,
          "maxValOfIntPart": 0,
          "decimalSeparator": "Comma",
          "minNrOfDecimals": 0,
          "maxNrOfDecimals": 0,
          "thousandSeparator": "None",
          "codePosition": "Before only",
          "currencySign": "Any",
          "currencyCodes": "EUR",
          "limitIntegerPart": false,
          "allowPlusSign": true,
          "allowMinusSign": true,
          "signIsRequired": false,
          "whitespaceAllowedAfterSign": true,
          "thousandSeparatorsAreRequired": false,
          "allowLeadingZeros": true,
          "requireIntegerPart": false,
          "allowExponent": false,
          "currencySignOrCodeRequired": true
        },
        "repeatInfo": {
          "repeat": "1"
        }
      }
    ]
  ],
  "generalRegexInfo": {
    "startRegexMatchAt": "Anywhere",
    "endRegexMatchAt": "Anywhere",
    "regexTarget": "Java"
  },
  "generateMethod": "samplesAndSemantics"
}
