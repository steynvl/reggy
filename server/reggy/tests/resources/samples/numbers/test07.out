Java
Pattern regex = Pattern.compile("(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*");
(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*
Perl
my $regex = /(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*/;
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
POSIX
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
Python
regex = re.compile(r'(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*')
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
JavaScript
const regex = /(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*/;
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
PHP
$regex = '/(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*/';
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
Golang
regex, _ := regexp.Compile("(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*")
(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*
Rust
let re = Regex::new(r"(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*").unwrap();
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
C#
var re = new Regex(@"(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*");
(?:[$€¥£]|EUR) *[\+-]\+? *0*\d*
Scala
val re = "(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*".r
(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*
Kotlin
val regex = Regex("(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*")
(?:[$€¥£]|EUR) *[\\+-]\\+? *0*\\d*